handsontable/src/plugins/pagination/pagination.js

```diff
@@ -868,6 +947,48 @@ export class Pagination extends BasePlugin {
     }
   }
 
+  /**
+   * Called when the focus is modified on the table. It sets the active context for the shortcuts.
+   *
+   * @param {'from_above' | 'from_below'} from The direction from which the focus was modified.
+   * @returns {boolean} Returns `false` to prevent the default focus behavior.
+   */
+  #onFocusTabNavigation(from) {
+    if (from === 'from_below') {
+      this.hot.getShortcutManager().setActiveContextName(SHORTCUTS_CONTEXT_NAME);
+
+      this.#focusController.focusLastElement();
+
+      return false;
+    }
+  }
+
+  /**
+   * Called when the focus is modified on the table. It sets the active context for the shortcuts.
+   *
+   * @param {'top' | 'bottom'} exitDirection The direction to which the focus was modified.
+   * @returns {boolean} Returns `false` to prevent the default focus behavior.
+   */
+  #onTableFocusExit(exitDirection) {
+    if (exitDirection === 'bottom') {
+      this.hot.deselectCell();
+      this.hot.getShortcutManager().setActiveContextName(SHORTCUTS_CONTEXT_NAME);
+
+      this.#focusController.focusFirstElement();
+
+      return false;
+    }
+  }
+
+  /**
+   * Called after the selection is made. It sets the active context for the shortcuts.
+   */
+  #onAfterSelection() {
+    if (this.hot.getShortcutManager().getActiveContextName() === SHORTCUTS_CONTEXT_NAME) {
+      this.hot.getShortcutManager().setActiveContextName('grid');
+    }
+  }
```

Comment:
I wonder if this "if" is necessary here? The hook is triggered after a selection is made on the table. It's not like it switches to `gird` anyway, so this hook is not needed?
```suggestion
  #onAfterSelection() {
    this.hot.getShortcutManager().setActiveContextName('grid');
  }
```